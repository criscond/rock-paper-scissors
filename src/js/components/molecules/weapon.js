import { createElement } from '../services';
import text from '../atoms/text';

const weapon = (item, onClick) => {
  const div = createElement({
    htmlTag: 'div',
    className: item.toLowerCase(),
    role: 'option'
  });

  div.appendChild(text({ htmlTag: 'h3', value: item }));
  div.addEventListener("click", () => onClick(item));

  return div;
}

export default weapon;
