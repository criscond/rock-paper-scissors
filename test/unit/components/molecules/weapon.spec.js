import weapon from '../../../../src/js/components/molecules/weapon';
describe('Weapon', () => {

  const weaponName = 'Paper';
  it('should create a element', () => {
    expect(weapon(weaponName)).toMatchSnapshot();
  });

  it('should create the weapon title', () => {
    const element = weapon(weaponName);

    expect(element.innerHTML).toContain('h3');
  });

  it('should dispatch a click event', () => {
    const onClickMock = jest.fn();

    const element = weapon(weaponName, onClickMock);
    const documentClickEvent = new Event('click');
    element.dispatchEvent(documentClickEvent);

    expect(onClickMock).toHaveBeenCalled();
  });
});
