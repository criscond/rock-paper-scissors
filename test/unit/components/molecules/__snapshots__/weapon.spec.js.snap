// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weapon should create a element 1`] = `
<div
  class="paper"
  role="option"
>
  <h3
    class=""
  >
    Paper
  </h3>
</div>
`;
